x/btcstaking/keeper/btc_delegations.go

```diff
@@ -95,6 +112,14 @@ func (k Keeper) addCovenantSigsToBTCDelegation(
 
 	k.setBTCDelegation(ctx, btcDel)
 
+	if err := ctx.EventManager().EmitTypedEvent(types.NewCovenantSignatureReceivedEvent(
```

Comment:
Not 100% sure about this -- do we need an event upon receiving each cov signature or upon receiving a covenant quorum?